File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        
        int n = intervals.length;
        Arrays.sort(intervals, (a, b) -> a[0]-b[0]);
        
        List<int[]> result = new ArrayList<>();
        
        for(int i=0; i<n; i++){
            
            if(result.size()==0 || intervals[i][0] > result.get(result.size()-1)[1]){
                result.add(intervals[i]);
            }
            else{
                result.get(result.size()-1)[1] = Math.max(result.get(result.size()-1)[1], intervals[i][1]);
            }
        }
        
        return result.toArray(new int[result.size()][]);
    }
}

File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        
        int n = intervals.size();
        sort(begin(intervals), end(intervals));
        
        
        vector<vector<int>> result;
        vector<int> temp = intervals[0]; // [1, 3]
        
        for(auto interval: intervals){
            
            if(interval[0] <= temp[1]){
                temp[0] = min(temp[0], interval[0]);
                temp[1] = max(temp[1], interval[1]);
            }
            else{
                result.push_back(temp);
                temp = interval;
            }
        }

        result.push_back(temp);
        return result;
    }
};